components/raftstore/src/store/worker/split_check.rs

```diff
@@ -187,8 +206,97 @@ where
         }
     }
 
+    fn approximate_check_bucket(
+        &self,
+        region: &Region,
+        host: &mut SplitCheckerHost<'_, E>,
+        bucket_ranges: Option<Vec<BucketRange>>,
+    ) {
+        let mut buckets: Vec<Bucket> = Vec::default();
+        let mut skip = false;
+        if let Some(ref bucket_ranges) = bucket_ranges {
+            skip = bucket_ranges.is_empty();
+            for bucket_range in bucket_ranges {
+                let mut bucket = region.clone();
+                bucket.set_start_key(bucket_range.0.clone());
+                bucket.set_end_key(bucket_range.1.clone());
+                let bucket_entry = match host.approximate_bucket_keys(&bucket, &self.engine) {
```

Comment:
How about replacing the parameter `Region` of approximate_bucket_keys with two slices of `u8`  for reducing clone?